limitador/src/storage/in_memory.rs

```diff
@@ -96,7 +96,7 @@ impl CounterStorage for InMemoryStorage {
         delta: i64,
         load_counters: bool,
     ) -> Result<Authorization, StorageErr> {
-        let mut limits_by_namespace = self.limits_for_namespace.write().unwrap();
+        let limits_by_namespace = self.limits_for_namespace.read().unwrap();
```

Comment:
This change introduces a race condition when we eventually update the counters [here](https://github.com/Kuadrant/limitador/pull/199/commits/9711f6338273ad4d44b47b6b8ee29e444b81d49f#diff-ebd607498f8416f1e94e7592efd8580d8dd6798df308dee8f7f7f7f78fe73c18R165-R173). By the time we get to our call `.update` on our `AtomicExpiringValue` references, their value could have already been updated by another thread. This is why we made `.update(...) -> u64`, i.e. return the value _at that updating moment_. But we now need to recheck we didn't pass the threshold... worst, if we did on say the second counter in our vec, what do we do with the increment to the first one?

So, tl;dr, I'm thinking lets leave grabbing the write lock for now and address the concurrency problem in a subsequent PR?